### ISSUE TITLE
Duplicates in automatically generated usernames

### ISSUE THREAD

**Issue by mk2301** [FoodCoopX/wirgarten-tapir#7]
At the moment, the username is generated `<firstname>.<lastname>`. 
So if 2 people have the same name, the second one can't register as the username must be unique (LDAP uid).

As the username is only really used for login, I would propose to use email for login instead. It better for the user as he/she doesn't need to remember a seperate username.

We need to allow change of the email address though, so probably we can't just use the email as unique id because it might be referenced somewhere. Maybe something like `hash(birthday, creationDate)` as uid, as those fields will never change.

**Comment by Theophile-Madet** [comment]
I'm not really convinced that users remember their email addresses more than their usernames, but I don't have a strong opinion on that. 

The main difficulty I think is that updating a username in LDAP is not straightforward. I don't think we can just update the `username` field of TapirUser. 

The `firstname.lastname` is only the default suggestion, it is just an editable string when creating the account with `CreateUserFromShareOwnerView`.  That view also gives an error if the username already exists. 
We'd need to check uniqueness for email addresses as well.

Having something readable as UID makes it easier for example to give someone extra rights on the LDAP admin page, though I guess relying on names can be error prone when several people have the same or similar names. 

I would not use the birthday in the LDAP UID, as there could be typing errors when creating the account and it could change later. What about just "account_[id]" with the TapirUser.id?

**Comment by mk2301** [comment]
Yes TapirUser.pk makes more sense, I will look into this soon and see how easy it is to do email login.

**Comment by gkirchner** [comment]
I just discussed with @oliveratfoodcoopx to follow a new approach for authentication with Keycloak and therefore at least put this issue on hold.
